.github/workflows/sync-branches-through-pr.yaml

```diff
@@ -1,47 +1,117 @@
+
+    - name: Set up Git
+      run: |
+        git config user.name "github-actions[bot]"
+        git config user.email "github-actions[bot]@users.noreply.github.com"
+
+    - name: Merge source branch into target with conflict resolution
+      id: merge
+      run: |
+        set -e
+        SOURCE_BRANCH="${{ github.event.inputs.source }}"
+        TARGET_BRANCH="${{ github.event.inputs.target }}"
+
+        git fetch origin ${TARGET_BRANCH}:${TARGET_BRANCH}
+        git checkout ${TARGET_BRANCH}
+
+        git fetch origin ${SOURCE_BRANCH}:${SOURCE_BRANCH}
+        git merge --no-commit origin/${SOURCE_BRANCH} || true
+
+        # Resolve conflicts for specific files
+        FILES=(
+          "components/notebook-controller/config/overlays/openshift/params.env"
+          "components/odh-notebook-controller/config/base/params.env"
+        )
+
+        for FILE in "${FILES[@]}"; do
+          if [[ -f "$FILE" && "$(git status --porcelain=v1 2>/dev/null | grep -c "$FILE")" -gt 0 ]]; then
+            echo "Resolving conflict for $FILE by keeping target branch version."
+            git checkout --ours "$FILE"
+            git add "$FILE"
+          fi
+        done
+
+        # Handle the Makefile in case there are other canges apart KF_TAG value
+        MAKEFILE="components/odh-notebook-controller/Makefile"
+        if [[ -f "$MAKEFILE" && "$(git status --porcelain=v1 2>/dev/null | grep -c "$MAKEFILE")" -gt 0 ]]; then
+          echo "Resolving conflicts in $MAKEFILE while preserving KF_TAG."
+          
+          # Extract KF_TAG from the target branch
+          KF_TAG_TARGET=$(cat "$MAKEFILE" | awk '
+            BEGIN { in_head = 0 }
+            /^<<<<<<< HEAD/ { in_head = 1; next }   # Enter HEAD block
+            /^=======/ { in_head = 0 }             # Exit HEAD block
+            in_head && /^KF_TAG \?=/ { print $0 }  # Extract full line
+          ' || echo "KF_TAG ?= main")
+
+
+          if [[ -z "$KF_TAG_TARGET" ]]; then
+            echo "KF_TAG not found in target branch; using a fallback value."
+            KF_TAG_TARGET="KF_TAG ?= main"
+          fi
+
+          # Process the conflicting Makefile
+          awk -v kf_tag="$KF_TAG_TARGET" '
+            BEGIN { conflict = 0 }
+            /^<<<<<<< / { conflict = 1; next }      # Start of conflict
+            /^=======/ { conflict = 2; next }      # Separator in conflict
+            /^>>>>>>> / { conflict = 0; next }     # End of conflict
+            conflict == 1 && /^KF_TAG \?=/ { next } # Ignore incoming KF_TAG
+            conflict == 2 && /^KF_TAG \?=/ { print kf_tag; next } # Use target KF_TAG
+            { print $0 }                           # Print other lines
+          ' "$MAKEFILE" > "$MAKEFILE.new"
+
+          # Replace original Makefile with resolved version
+          mv "$MAKEFILE.new" "$MAKEFILE"
+          echo "Preserved KF_TAG as: $KF_TAG_TARGET"
+
+          # Stage the resolved Makefile
+          git add "$MAKEFILE"
+        fi
```

Comment:
Wouldn't it be easier to use `theirs` version, and then it would not be necessary to have this much logic to handle the Makefile? Alternatively, would it be possible to put the `KF_TAG` value into a separate file, so that both this logic and the logic that updates the digest there could be much simplified?

Reply:
Thank you for the review! I agree with your feedback. I’ve updated the logic to consume the `KF_TAG` from an external file instead of maintaining the extensive logic directly. I initially aimed to minimize changes to the existing code, but this approach is much cleaner and more maintainable.